File: library/src/main/java/com/github/javiersantos/piracychecker/enums/PirateApp.java
package com.github.javiersantos.piracychecker.enums;

import android.support.annotation.NonNull;
import android.text.TextUtils;

public class PirateApp {
    private String name;
    private String[] pack;
    private AppType type;

    public PirateApp(@NonNull String name, @NonNull String[] pack, @NonNull AppType type) {
        this.name = name;
        this.pack = pack.clone();
        this.type = type;
    }

    public PirateApp(@NonNull String name, @NonNull String appPackage, @NonNull AppType type) {
        this.name = name;
        this.pack = TextUtils.split(appPackage, "");
        this.type = type;
    }

    public PirateApp(@NonNull String name, @NonNull String appPackage) {
        this(name, appPackage, AppType.OTHER);
    }

    public String getName() {
        return name;
    }

    @Deprecated
    public String[] getPack() {
        return pack;
    }

    public String getPackage() {
        StringBuilder sb = new StringBuilder();
        for (String s : pack) {
            sb.append(s);
        }
        return sb.toString();
    }

    public AppType getType() {
        return type;
    }
}

File: library/src/main/java/com/github/javiersantos/piracychecker/enums/PirateApp.kt
package com.github.javiersantos.piracychecker.enums

import android.text.TextUtils

class PirateApp(name: String, pack: Array<String>, type: AppType = AppType.OTHER) {
    var name: String? = null
        private set
    var type: AppType? = null
        private set
    private var pack: Array<String>? = null
    
    init {
        this.name = name
        this.pack = pack.clone()
        this.type = type
    }
    
    @Deprecated("Deprecated in favor of packageName", ReplaceWith("packageName"))
    val `package`: String
        get() = packageName
    
    val packageName: String
        get() {
            val sb = StringBuilder()
            pack?.forEach { sb.append(it) }
            return sb.toString()
        }
    
    @JvmOverloads
    constructor(name: String, appPackage: String, type: AppType = AppType.OTHER) :
        this(name, TextUtils.split(appPackage, ""), type)
}